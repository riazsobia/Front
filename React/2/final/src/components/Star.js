import "../styles/star.css";

function Star(props) {
  return (
    <div
      className="crop"
      style={{ width: props.rating ? (90 / 5) * props.rating : 0 }}
      title={props.rating}
      onClick={() => props.onClick(props.rating)}
    >
      <div className="stars">
        <span className="fa-solid fa-star"></span>
        <span className="fa-solid fa-star"></span>
        <span className="fa-solid fa-star"></span>
        <span className="fa-solid fa-star"></span>
        <span className="fa-solid fa-star"></span>
      </div>
    </div>
  );
}

export default Star;
