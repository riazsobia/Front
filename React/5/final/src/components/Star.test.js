import React from "react";
import renderer from "react-test-renderer";
import Star from "./Star";

it("should create 5 stars if rating is 5", () => {
  const tree = renderer.create(<Star rating={5} />);

  expect(tree).toMatchSnapshot();
});
