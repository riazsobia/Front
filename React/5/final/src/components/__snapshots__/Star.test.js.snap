// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create 5 stars if rating is 5 1`] = `
<div
  className="crop"
  onClick={[Function]}
  style={
    Object {
      "width": 90,
    }
  }
  title={5}
>
  <div
    className="stars"
  >
    <span
      className="fa-solid fa-star"
    />
    <span
      className="fa-solid fa-star"
    />
    <span
      className="fa-solid fa-star"
    />
    <span
      className="fa-solid fa-star"
    />
    <span
      className="fa-solid fa-star"
    />
  </div>
</div>
`;
